rllib/algorithms/algorithm_config.py

```diff
@@ -1096,28 +1290,128 @@ def multi_agent(
                 error=True,
             )
+
         if count_steps_by is not None:
+            if count_steps_by not in ["env_steps", "agent_steps"]:
+                raise ValueError(
+                    "config.multi_agent(count_steps_by=..) must be one of "
+                    f"[env_steps|agent_steps], not {count_steps_by}!"
+                )
             self.count_steps_by = count_steps_by
 
+        # Policies given as set/list/tuple (of PolicyIDs) -> Setup each policy
+        # automatically via empty PolicySpec (will make RLlib infer observation- and
+        # action spaces as well as the Policy's class).
+        if isinstance(self.policies, (set, list, tuple)):
+            self.policies = {pid: PolicySpec() for pid in self.policies}
+
+        # Check each defined policy ID and unify its spec.
+        for pid, policy_spec in self.policies.copy().items():
+            # Make sure our Policy ID is ok.
+            validate_policy_id(pid, error=False)
+
+            # Policy IDs must be strings.
+            if not isinstance(pid, str):
+                raise KeyError(
+                    f"Policy IDs must always be of type `str`, got {type(pid)}"
+                )
+
+            # Convert to PolicySpec if plain list/tuple.
+            if not isinstance(policy_spec, PolicySpec):
+                # Values must be lists/tuples of len 4.
+                if not isinstance(policy_spec, (list, tuple)) or len(policy_spec) != 4:
+                    raise ValueError(
+                        "Policy specs must be tuples/lists of "
+                        "(cls or None, obs_space, action_space, config), "
+                        f"got {policy_spec}"
+                    )
+                self.policies[pid] = PolicySpec(*policy_spec)
+
+            # Config is None -> Set to {}.
+            if self.policies[pid].config is None:
+                self.policies[pid].config = {}
+            # TODO: Switch from dict to AlgorithmConfigOverride, once available.
+            # Config not a dict.
+            elif not isinstance(self.policies[pid].config, dict):
+                raise ValueError(
+                    f"Multi-agent policy config for {pid} must be a dict, "
+                    f"but got {type(self.policies[pid].config)}!"
+                )
+
+        # Is this a multi-agent setup? True, iff DEFAULT_POLICY_ID is only
+        # PolicyID found in policies dict.
+        self._is_multi_agent = (
+            len(self.policies) > 1 or DEFAULT_POLICY_ID not in self.policies
+        )
+
+        self._set_ma_legacy_dict()
+
         return self
 
+    def is_multi_agent(self) -> bool:
+        """Returns whether this config specifies a multi-agent setup.
+
+        Returns:
+            True, if a) >1 policies defined OR b) 1 policy defined, but its ID is NOT
+            DEFAULT_POLICY_ID.
+        """
+        return self._is_multi_agent
+
+    def _set_ma_legacy_dict(self):
```

Comment:
Helper method to backward-fix the "multiagent" key (only if users still access this).